# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/alsa-utils/hotfix-glibc-2.37.patch
# Copyright (C) 2023 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- alsa-utils-1.2.9/alsactl/init_sysdeps.c.vanilla	2023-08-04 23:14:42.829273016 +0200
+++ alsa-utils-1.2.9/alsactl/init_sysdeps.c	2023-08-04 23:22:11.137281323 +0200
@@ -17,6 +17,8 @@
  */
 
 #if defined(__GLIBC__) && !(defined(__UCLIBC__) && defined(__USE_BSD))
+#if defined __GLIBC__ && ((__GLIBC__ >= 2) && (__GLIBC_MINOR__ >= 37))
+#else
 static size_t strlcpy(char *dst, const char *src, size_t size)
 {
 	size_t bytes = 0;
@@ -62,4 +62,5 @@
 	*q = '\0';
 	return bytes;
 }
+#endif
 #endif /* __GLIBC__ */
